doc/develop/api/api_conventions.rst

```diff
@@ -0,0 +1,49 @@
+.. _api_conventions.rst:
+
+API Conventions
+###############
+
+Zephyr provides a few programming interfaces that users can use for creating their applications.
+Zephyr's API space includes the following areas:
+- C programming interface (function prototypes, structures, and macros), provided in C header files
+- Configuration system (Kconfig)
+- Device driver model (Devicetree)
+
+Other areas, such as the build system (CMake), may also be seen as Zephyr's API. However, any area
+that is not explicitly listed herein is not considered Zephyr's API as of today.
+
+API classification
+******************
+The files that define the Zephyr APIs may contain symbols intended for different usage. The intended
+API usage is designated by the API class. Zephyr APIs are classified as:
+
+- private - these APIs are intended for use in-tree within boundary of a software component such as
+            driver or subsystem. Private APIs are not subject to :ref:`api_lifecycle`. Therefore can
+            be changed or removed at any time.
+- internal - a class of APIs that serve as the interface between two or more low level areas.
+- public - a class of APis intended for use in-tree and out-of-tree. The public APIs are subject to
+           the :ref:`api_lifecycle`. This is the class that most users should use.
+
+The following sections provide guidelines on how to identify the class of an API depending on its
+type.
+
+Header files contents
+=====================
+- private - symbols prefixed with `z_` are private.
+- internal - <TBD>
+- public - functions and data types declared in header files located in `include/zephyr`.
+
+Due to historical reasons some APIs prefixed with `z_` are public. These are:
+- <TBD>
+
+Kconfig symbols
+===============
+- private - not applicable
+- internal - symbols that are promptless and hidden
```

Comment:
'internal' here should have the same scope as 'internal' does for APIs.  I think this means that symbols that are promptless and hidden are "private" rather than "internal".

Reply:
Agree.